### ISSUE TITLE
Emulate multi-column assigment SET clauses in UPDATE statements, where this is not supported

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#1913]
There are some indications in this Stack Overflow question about how this sort of `UPDATE` could be simulated in those databases that don't natively support it:

``` sql
-- SQL standard:
UPDATE DEST D SET (AAA,BBB) = (
    SELECT MAX(Z.AAA), MAX(Z.BBB) FROM OTHER O WHERE O.ID = D.ID
)

-- Possible emulation:
UPDATE DEST D
    SET AAA = (SELECT MAX(Z.AAA) FROM OTHER O WHERE O.ID = D.ID),
        BBB = (SELECT MAX(Z.BBB) FROM OTHER O WHERE O.ID = D.ID)

-- If MERGE and OUTER APPLY / LATERAL is available
MERGE INTO DEST D
USING (
  SELECT U.* FROM
  DEST D
  OUTER APPLY (
    SELECT MAX(Z.AAA), MAX(Z.BBB) FROM OTHER O WHERE O.ID = D.ID
  ) AS U (U1, U2)
) AS S
ON (1 = 1)
WHEN MATCHED THEN UPDATE SET AAA = S.U1, BBB = S.U2
```

See more here:
http://stackoverflow.com/q/585620/521799

Some dialects (such as SQL Server) know an `UPDATE .. FROM` clause:
http://stackoverflow.com/q/2334712/521799

This depends on #1583

**Comment by lukaseder** [comment]
Duplicate of https://github.com/jOOQ/jOOQ/issues/10523 and https://github.com/jOOQ/jOOQ/issues/10522
